superflore/test_integration/gentoo/build_base.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from docker.errors import ContainerError
+from superflore.docker import Docker
+from superflore.utils import err
+from superflore.utils import info
+from superflore.utils import ok
+
+
+class GentooBuilder:
+    def __init__(
+        self, image_owner='allenh1', image_name='ros_gentoo_base'
+    ):
+        self.container = Docker()
+        self.container.login()
+        self.container.pull(image_owner, image_name)
+        self.package_list = dict()
+
+    def add_target(self, ros_distro, pkg):
+        # TODO(allenh1): it might be nice to add a Python3 target
+        # in case we want to test both.
+        self.package_list['ros-%s/%s' % (ros_distro, pkg)] = 'unknown'
+
+    def run(self):
+        # TODO(allenh1): add the ability to check out a non-master
+        # branch of the overlay (for CI).
+        info('testing gentoo package integrity')
+        for pkg in sorted(self.package_list.keys()):
+            self.container.add_bash_command('emaint sync -r ros-overlay')
+            self.container.add_bash_command('emerge %s' % pkg)
+            self.conainter.login()
```

Comment:
spelling conainter

Reply:
Fixed. That line didn't need to be there in the first place, actually.